dis_snek/api/voice/player.py

```diff
@@ -0,0 +1,145 @@
+import asyncio
+import logging
+import threading
+import time
+from asyncio import AbstractEventLoop
+from typing import Optional, TYPE_CHECKING
+
+from dis_snek.client.const import logger_name
+from dis_snek.api.voice.audio import BaseAudio, AudioVolume
+from dis_snek.api.voice.opus import Encoder
+
+if TYPE_CHECKING:
+    from dis_snek.models.snek.active_voice_state import ActiveVoiceState
+__all__ = ["Player"]
+
+log = logging.getLogger(logger_name)
+
+
+class Player(threading.Thread):
```

Comment:
Just curious, have you done timing on the voice here? How long does it take to compute (ie. time blocking the event loop)?

Reply:
It wasnt blocking the event loop that was the problem, it was voice not firing consistently enough when running as an asyncio task. Not to mention, with discord's voice implimentation, you **must** send the payload on time, or you will experience stuttering. As such a threaded solution is prefered over an asyncio task.